Sources/Caching/FileRepository.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  FileRepository.swift
+//
+//  Created by Jacob Zivan Rakidzich on 8/13/25.
+
+import Foundation
+
+/// A file cache
+@_spi(Internal) public class FileRepository: @unchecked Sendable {
```

Comment:
Nit: I might consider making `FileRepository` conform to a `FileRepository` protocol so it's easier to mock out in tests in the future